tests/test_process.py

```diff
@@ -202,6 +203,168 @@ def teardown_method(self):
+        ):
+            return df.assign(test_col=2)
+
+        return _standard_transform_function
+
+    @pytest.fixture(scope="function", autouse=True)
+    def special_transform_function(self):
+        """mock transform function that takes test_threshold as an argument"""
+
+        def _mock_transform_with_args(df: pd.DataFrame, test_threshold: int):
+            return df.assign(new_key=1)
+
+        return _mock_transform_with_args
+
+    @pytest.mark.parametrize(
+        "example_config_with_custom_transform,expectation",
+        [
+            (
+                {
+                    "neuropath_corr": {
+                        "files": [
+                            {"name": "test_file_1", "id": "syn1111111", "format": "csv"}
+                        ],
+                        "final_format": "json",
+                        "provenance": ["syn1111111"],
+                        "destination": "syn1111113",
+                        "gx_enabled": False,
+                        "custom_transformations": {
+                            "mock_transform_function_false": {
+                                "test_threshold": 1,
+                                "test_p_value": 2,
+                            }
+                        },
+                    }
+                },
+                pytest.raises(AttributeError),
+            ),
+            (
+                {
+                    "neuropath_corr": {
+                        "files": [
+                            {"name": "test_file_1", "id": "syn1111111", "format": "csv"}
+                        ],
+                        "final_format": "json",
+                        "provenance": ["syn1111111"],
+                        "destination": "syn1111113",
+                        "gx_enabled": False,
+                        "custom_transformations": "mock_transform_function_false",
+                    },
+                },
+                pytest.raises(AttributeError),
+            ),
+            (
+                {
+                    "neuropath_corr": {
+                        "files": [
+                            {"name": "test_file_1", "id": "syn1111111", "format": "csv"}
+                        ],
+                        "final_format": "json",
+                        "provenance": ["syn1111111"],
+                        "destination": "syn1111113",
+                        "gx_enabled": False,
+                        "custom_transformations": 1,
+                    },
+                },
+                pytest.raises(AttributeError),
+            ),
+        ],
+    )
+    def test_apply_invalid_custom_transformations(
+        self,
+        example_config_with_custom_transform: Dict[str, Any],
+        expectation: Exception,
+    ) -> None:
+        """Test that invalid custom transformations raise an error."""
+        # disable the class level patcher
+        mock.patch.stopall()
```

Comment:
@BWMac @BryanFauble In the setup_method, there's a patch for `process.apply_custom_transformations`. But I need to test this function, so I added `mock.patch.stopall()` here to stop all the patches for this test. Is that okay? I tried `self.patch_custom_transform.stop()` but it didn't stop the patch

Reply:
You should be testing process.apply_custom_transformations in it's own class `TestApplyCustomTransformations` instead which would obviate this issue.